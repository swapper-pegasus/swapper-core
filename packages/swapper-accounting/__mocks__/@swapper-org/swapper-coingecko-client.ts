import {
  ResponseGetTokenPrice,
  ResponseGetTokens
} from '@swapper-org/swapper-coingecko-client'
export class SwapperCoingeckoClient {
  public getTokensPriceInUsd (): Promise<ResponseGetTokenPrice> {
    const response: ResponseGetTokenPrice = { bitcoin: { usd: '100000' }, ethereum: { usd: '1000' } }
    return new Promise((resolve) => resolve(response))
  }

  public getTokenDataBySymbol (): Promise<ResponseGetTokens> {
    const response: ResponseGetTokens = [{ id: 'ethereum', name: 'Ethereum', symbol: 'eth' }, { id: 'bitcoin', name: 'Bitcoin', symbol: 'btc' }]
    return new Promise((resolve) => resolve(response))
  }
}
