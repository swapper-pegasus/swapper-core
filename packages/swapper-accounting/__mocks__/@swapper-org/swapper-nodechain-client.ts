import {
  ResponseGetAddressesBalance
} from '@swapper-org/swapper-nodechain-client'

export class SwapperNodechainEthClient {
  public getAddressesBalance (): Promise<ResponseGetAddressesBalance> {
    const response: ResponseGetAddressesBalance = {
      id: 1,
      jsonrpc: 'getAddressesBalance',
      result: [{
        address: '0xABC',
        balance: {
          confirmed: '0xA',
          unconfirmed: '0x0'
        }
      }]
    }
    return new Promise((resolve) => resolve(response))
  }
}

export class SwapperNodechainBtcClient {
  public getAddressesBalance (): Promise<ResponseGetAddressesBalance> {
    const response: ResponseGetAddressesBalance = {
      id: 1,
      jsonrpc: 'getAddressesBalance',
      result: [{
        address: 'abc',
        balance: {
          confirmed: '0x1',
          unconfirmed: '0x0'
        }
      }]
    }
    return new Promise((resolve) => resolve(response))
  }
}
