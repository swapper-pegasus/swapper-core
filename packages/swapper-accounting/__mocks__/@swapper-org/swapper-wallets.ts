import {
  WalletBalance
} from '@swapper-org/swapper-wallets'
class EthWallet {
  public getBalance (): Promise<WalletBalance> {
    const response: WalletBalance = { confirmed: '0x1', unconfirmed: '0x0' }
    return new Promise((resolve) => resolve(response))
  }
}

export { EthWallet }
