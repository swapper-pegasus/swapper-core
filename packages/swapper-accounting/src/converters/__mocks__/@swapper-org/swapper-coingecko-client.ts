export class SwapperCoingeckoClient {
  public getTokensPriceInUsd = jest.fn().mockResolvedValue({})
  public getTokenDataBySymbol = jest.fn().mockResolvedValue({})
}
