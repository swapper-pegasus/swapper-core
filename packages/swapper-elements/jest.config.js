// eslint-disable-next-line @typescript-eslint/no-var-requires
const config = require('../../jest.config')

module.exports = {
  ...config,
  moduleFileExtensions: ['js', 'ts', 'tsx', 'json']
}
