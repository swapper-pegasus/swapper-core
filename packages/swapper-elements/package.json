{
  "name": "@swapper-org/swapper-elements",
  "version": "0.0.38",
  "main": "lib/index.js",
  "types": "lib/index.d.ts",
  "module": "lib/index.js",
  "license": "(GPL-3.0-or-later OR AGPL-3.0-or-later)",
  "scripts": {
    "build": "yarn build:cjs",
    "build:dev": "yarn tsc -b",
    "build:cjs": "yarn tsc --project ./tsconfig.build.json",
    "test": "jest --coverage"
  },
  "repository": {
    "type": "git",
    "url": "ssh://git@github.com:swapper-org/swapper-core.git",
    "directory": "packages/swapper-elements"
  },
  "publishConfig": {
    "registry": "https://npm.pkg.github.com/"
  },
  "devDependencies": {
    "@testing-library/jest-dom": "^5.16.1",
    "@testing-library/react": "^12.1.2",
    "@types/jest": "^27.0.3",
    "@types/react": "^17.0.26",
    "@types/react-dom": "^17.0.9",
    "@types/react-test-renderer": "^17.0.1",
    "jest": "^27.4.3",
    "react": "^17.0.2",
    "react-dom": "^17.0.2",
    "react-test-renderer": "^17.0.2",
    "ts-jest": "^27.1.0",
    "typescript": "^4.4.3"
  },
  "peerDependencies": {
    "react": "^17.0.2",
    "react-dom": "^17.0.2"
  },
  "devEngines": {
    "node": ">=14.x"
  },
  "dependencies": {
    "@heroicons/react": "^1.0.5"
  }
}
