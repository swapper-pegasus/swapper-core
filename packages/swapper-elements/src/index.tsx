export * from './components/Input'
export { Button } from './components/Button'
export { Modal } from './components/Modal'
export { Image } from './components/Image'
export { Alert } from './components/Alert'
export { Icon } from './components/Icon'
