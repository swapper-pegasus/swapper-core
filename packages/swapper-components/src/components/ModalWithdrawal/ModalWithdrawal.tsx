import React, { useState, useEffect } from 'react'
import { Modal, Button, Input, Alert } from '@swapper-org/swapper-elements'
import { WalletBalance } from '@swapper-org/swapper-wallets'
import BN from 'bignumber.js'
export type Props = {
  address: string,
  tokenName: string,
  balance: WalletBalance
  isOpen: boolean,
  error?: string,
  isLoading?: boolean,
  onClose: () => void;
  onSend: (address: string, amount: string) => void;
};

export function ModalWithdrawal ({ tokenName, isLoading, error, address, balance, isOpen, onClose, onSend }: Props) {
  const [amount, setAmount] = useState<string>('')
  const [addressToSend, setAddressToSend] = useState<string>('')
  const [amountError, setAmountError] = useState('')

  useEffect(() => {
    new BN(amount).comparedTo(new BN(balance.confirmed)) === 1 ? setAmountError('Not enougth funds.') : setAmountError('')
  }, [amount, balance])

  return (
    <Modal isOpen={isOpen} onClose={() => onClose()}>
      <div className='w-full' data-testid='withdrawal-modal'>
        {
          isLoading
            ? <p>Loading...</p>
            : <>
                <div className="flex flex-col justify-center items-center min-w-full">
                  <Alert
                    type='warning'
                    description={`Sending ${tokenName}.`}
                  />
                  {
                    error && <Alert
                      type='error'
                      description={error}
                    />
                  }
                  <div className="mt-2 min-w-full">
                    <p className='font-medium'>Your address: {address}</p>
                  </div>
                  <div className="mt-2 min-w-full">
                    <Input
                      placeholder="Send to this address"
                      label="Send to"
                      name="address"
                      value={addressToSend}
                      onChange={(value) => setAddressToSend(value)}
                    />
                  </div>
                  <div className="mt-2 min-w-full">
                    <Input
                      placeholder="Amount"
                      label="Amount"
                      name="amount"
                      value={amount}
                      type="number"
                      onChange={(value) => setAmount(value)}
                      error={amountError}
                    />
                  </div>
                </div>
                <div className="bg-white pt-3 py-3 text-right flex justify-between">
                  <div className='mr-4'>
                    <Button dataTestId='modal-cancel-button' onClick={() => onClose()}>
                      <span>Cancel</span>
                    </Button>
                  </div>
                  <Button dataTestId='modal-withdrawal-button' disabled={!!amountError} onClick={() => {
                    onSend(addressToSend, amount)
                  }}>
                    <span>Continue</span>
                  </Button>
                </div>
              </>
        }
      </div>
    </Modal>
  )
}
