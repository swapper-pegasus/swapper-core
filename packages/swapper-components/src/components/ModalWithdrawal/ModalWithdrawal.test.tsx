import React, { useState } from 'react'
import '@testing-library/jest-dom'
import { ModalWithdrawal, Props } from './ModalWithdrawal'
import { Button } from '@swapper-org/swapper-elements'
import { render, fireEvent } from '@testing-library/react'

const TestComponent = (props: Props) => {
  const [isOpen, setIsOpen] = useState(false)
  return (
        <>
            <div id='modalContainer'></div>
            <Button dataTestId='button-open-modal' onClick={() => setIsOpen(true)}></Button>
            <ModalWithdrawal {...props} isOpen={isOpen} />
        </>
  )
}

const defaultProps = {
  address: '0xABC',
  tokenName: 'Ethereum',
  balance: { confirmed: '12.23123', unconfirmed: '0' },
  isOpen: true,
  onClose: jest.fn(),
  onSend: jest.fn()
}

function setUpRender (props: object) {
  const finalProps = {
    ...defaultProps,
    ...props
  }
  return render(
    <TestComponent {...finalProps}/>
  )
}

describe('ModalWithdrawal', () => {
  it('Send on continue', () => {
    const mockOnSend = jest.fn()
    const component = setUpRender({ onSend: mockOnSend })
    fireEvent.click(component.getByTestId('button-open-modal'))
    fireEvent.change(component.getByTestId('address-input'), { target: { value: '0xAAAA' } })
    fireEvent.change(component.getByTestId('amount-input'), { target: { value: '1.2' } })
    fireEvent.click(component.getByTestId('modal-withdrawal-button'))
    expect(mockOnSend).toHaveBeenCalledWith('0xAAAA', '1.2')
  })
  it('Close on cancel', () => {
    const mockOnClose = jest.fn()
    const component = setUpRender({ onClose: mockOnClose })
    fireEvent.click(component.getByTestId('button-open-modal'))
    fireEvent.change(component.getByTestId('address-input'), { target: { value: '0xAAAA' } })
    fireEvent.change(component.getByTestId('amount-input'), { target: { value: '1.2' } })
    fireEvent.click(component.getByTestId('modal-cancel-button'))
    expect(mockOnClose).toHaveBeenCalledWith()
  })
  it('Not enougth funds', () => {
    const component = setUpRender({})
    fireEvent.click(component.getByTestId('button-open-modal'))
    fireEvent.change(component.getByTestId('address-input'), { target: { value: '0xAAAA' } })
    fireEvent.change(component.getByTestId('amount-input'), { target: { value: '123' } })
    expect(component.getByTestId('error-amount-input')).toHaveTextContent('Not enougth funds.')
  })
  it('Close press ESC', () => {
    const mockOnClose = jest.fn()
    const component = setUpRender({ onClose: mockOnClose })
    fireEvent.click(component.getByTestId('button-open-modal'))
    fireEvent.keyDown(component.getByRole('dialog'), { key: 'esc', keyCode: 27 })
    expect(mockOnClose).toHaveBeenCalledTimes(1)
  })
})
