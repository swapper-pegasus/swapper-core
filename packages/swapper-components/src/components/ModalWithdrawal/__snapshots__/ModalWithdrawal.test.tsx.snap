// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalWithdrawal Loading state 1`] = `
<body>
  <div>
    <div
      id="modalContainer"
    >
      <div
        aria-labelledby="modal-title"
        aria-modal="true"
        class="fixed inset-0 overflow-y-auto w-2/6 m-auto"
        role="dialog"
      >
        <div
          class="flex items-center justify-center min-h-screen pt-4 px-4 pb-20 text-center"
        >
          <div
            aria-hidden="true"
            class="fixed inset-0 bg-gray-500 bg-opacity-75 transition-opacity"
          />
          <span
            aria-hidden="true"
            class="hidden"
          >
            â€‹
          </span>
          <div
            class="inline-block align-bottom bg-white rounded-lg text-left overflow-hidden shadow-xl transform transition-all p-6"
          >
            <div
              class="bg-white"
            >
              <div
                class="w-full"
                data-testid="withdrawal-modal"
              >
                <p>
                  Loading...
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <button
      class="disabled:cursor-not-allowed disabled:opacity-20 bg-primary rounded-full hover:bg-primary-dark text-white text-sm px-8 py-2"
      data-testid="button-open-modal"
    />
  </div>
</body>
`;
